### PR TITLE
Show error message if TE task is not accessible to the user

### PR DESCRIPTION
**Pull request by DadImScared**
https://trello.com/c/IDkOo1Kg/286-show-error-message-if-te-task-is-not-accessible-to-the-user

### CODE REVIEW

File: frontend/src/views/TE/Task.vue [hunk 1]
```diff
@@ -187,6 +192,7 @@ export default class Task extends Mixins<TaskMixin, FormatTime>(
   task: any = {};
   isFetchingTask = true;
   isCoordinator = false;
+  fetchTaskError = false;
```
Comment (anantakrishna) [inline frontend/src/views/TE/Task.vue]: Please read https://itnext.io/managing-local-state-in-vue-js-components-d5d92fb432 and decide which of two approaches is suitable for this component. I opt for “String” approach as the states are mutually exclusive.

File: frontend/src/views/TE/Task.vue [hunk 2]
```diff
@@ -174,6 +174,12 @@ import TaskMixin from "@/components/TE/TaskMixin";
 import FormatTime from "@/mixins/FormatTime";
 import VersionDownloadLink from "@/components/TE/VersionDownloadLink.vue";
 
+enum TaskState {
```
Comment (anantakrishna) [inline frontend/src/views/TE/Task.vue]: I'd say it's not a task state but the page state. Did other pages there will be also state enum, and no need to prefix it with a specific term. There will be usually only one enum per page.

File: frontend/src/views/TE/Task.vue [hunk 3]
```diff
@@ -190,9 +196,9 @@ export default class Task extends Mixins<TaskMixin, FormatTime>(
   FormatTime
 ) {
   task: any = {};
-  isFetchingTask = true;
+  TaskState = TaskState;
```
Comment (anantakrishna) [inline frontend/src/views/TE/Task.vue]: Can we avoid this duplication?

Comment (DadImScared) [reply frontend/src/views/TE/Task.vue]: To use the enum in the template I think it needs to be declared in the class.

**Pull request merged**